### ISSUE TITLE
Support of multi OrientDB environment

### ISSUE THREAD

**Issue by PhantomYdn** [PhantomYdn/Orienteer#125]

**Comment by PhantomYdn** [comment]
@krichter722  :
`It'd be great if the database connection would be configurable (including a switch for connecting to the (or even multiple) embedded instance(s)). I can't think of any other option I'm missing to configurable at runtime in the web frontend, but I formulated the title as abstract as possible.`

**Comment by PhantomYdn** [comment]
Idea is the following:

There is master orientdb database with whole set of users who can have access to an environment. Every user can have one to many child databases. Databases can be shared among some subset of users (group rights). User can transparently navigate to a child db after initial login.

**Comment by krichter722** [comment]
I'm far from being an expert of orientdb features. Is a "child database" a special feature? What should I think of? A cluster sharing deduplication, a nested database (can't imagine that)?

Regardless of the question above, I'd say managing database instances (ownership, location, etc.) with the client is overkill if these matters are managed by the database itself. Instead provide assistence for database creation, moving and ownership/permission management - think of a `postgresql` server and a `pgadmin` client; I'd do the separation of responsibility exactly like those two. Implementing a mapping between Orienteer users database users shouldn't be underestimated, although you know the orientdb code quiet well. Managing different database with different separate connection parameters (which can be stored in Orienteer) is much easier and allows the same features as "transparently navigate to a child db".

The idea of a master database is then obsolete. If you use an orientdb instance to store Orienteer settings (haven't checked how it's actually done, but it might change anyway), I wouldn't expose it to the user at all.

**Comment by krichter722** [comment]
Any news on that? I'm willing to take a look into Orienteer development again.

**Comment by PhantomYdn** [comment]
Hi,

The main idea of this project is to make a [data warehouse](http://en.wikipedia.org/wiki/Data_warehouse) (actually [data lake](http://en.wikipedia.org/wiki/Data_lake)), so no "dropdown" list for selection of DBs as it commonly is for DB cliens, was not assumed.

Also I have idea to build a SaaS for Orienteer. To be able to implement that, I'm going to use a concept that I mentioned before: child DB - are OrientDB instancies ordered by clients and parent DB is for management of whole SaaS services.

If you can also change Orienteer in a way as you want to have it: that will be cool. So Orienteer will cover interest of 2 categories: 1) clients who want to have a Data Lake for their business 2) clients who just need to have Data Management tool

**Comment by krichter722** [comment]
> If you can also change Orienteer in a way as you want to have it: that will be cool. So Orienteer will cover interest of 2 categories: 1) clients who want to have a Data Lake for their business 2) clients who just need to have Data Management tool

I have to admit I didn't get in touch with the terms "Data warehouse" and "Data lake" so far (learned the exact definition today), but I think both approaches 1) and 2) are perfectly compatible and code can be widely reused. My comment 4 above applies to 2) then and the idea of master and child databases is perfect for 1).

**Comment by PhantomYdn** [comment]
Yes, but there is 1 small thing which limits using Orienteer as OrientDB client: to be able operate as Data Warehouse Orienteer should change schema of a DB for its own needs. So after referencing Orienteer to arbitrary DB - that DB will be changed.
